var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -291,6 +293,10 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("spglib", when="+spglib")
 
+    depends_on("dftd4@3.6.0:", when="+dftd4")
+    # DFTD4 support is broken in CP2K @:2025.1 with CMake
+    conflicts("+dftd4", when="@:2025.1 build_system=cmake")
```

Comment:
We could create a patch for 2025.1 and below instead of explicitly setting it up as conflict.

Reply:
I have a patch almost ready. I went for this because I was thinking I needed this in Spack before adding CI tests on CP2K side, but now remembered CP2K does not use Spack to install itself, only dependencies. I'll finish off the CP2K PR, then update this PR with the patch. I'm not a big fan of patches, but in this case I think it might be worth it (at least for `2025.1` as you suggest), so that we can deploy it to our users.

Reply:
I'll split the `dftd4` change in a separate PR, because that is needed. With `dftbd@3.5.0` I was unable to run, while `dftd4@3.6.0` (the version used in CP2K toolchain) it works well.

Reply:
Using `build_system=cmake` should work with older versions too. I removed the conflict.